### ISSUE TITLE
Update single note via REST API

### ISSUE THREAD

**Issue by cmuench** [zadam/trilium#646]
I am working on a command for ULauncher to add new entries to the current day note.
To archive this, I use the REST api.
With `/api/date-notes/date/:date` I am able to get the right note meta data.
With `/api/notes/:noteid` I am able to fetch the note data.
What's missing is the endpoint to update a single note without a CSRF check.
Is there a plan to extend the api so that it can be used without the desktop or web application?
This will open the tool for a wide range of applications.

**Comment by xeor** [comment]
I'm in the work on figuring out the same.. A cli or more api options would be a very nice addition. I found a way to do it using the app (https://www.reddit.com/r/Trilium/comments/d2vx09/app_and_customrequesthandler/), but I think the better way in your case would be to just make an custom-request-handler.

I got some code to get past the CSRF check, but then I started getting `Error: No data key!` exceptions because of the way trilium handles session and encryption.

All in all, I decided to try another route, and that is to small custom-request-handler's for different functionality, then a cli that talks to them. Then I can have different api keys for different functions (example, an append-only mode), and I don't haveto fight against internals and undocumented api's.
